### ISSUE TITLE
Deserialization of 2.0 indexes broken in next

### ISSUE THREAD

**Issue by danielmewes** [rethinkdb/rethinkdb#4169]
It crashes the server with a range error. Investigating.

**Comment by danielmewes** [comment]
```
The first bad commit could be any of:
4d0a80e56eac6a7d9baf15bae762b8304a4984f7
a54ac14c86d1a8347cdceb3d575ada0c547d6a7b
We cannot bisect more!
```

I think this is because `wire_func_t` used to have a protobuf backtrace, and now it has a `backtrace_id_t`.

I imagine we can fix this by adding a 2.1 `cluster_version_t` and modifying `archive_result_t deserialize(read_stream_t *s, wire_func_t *wf)` in `wire_func.cc` to somehow translate the protob_t<Backtrace> to a backtrace_id_t?

@Tryneus Could you take a look at this? You'll probably know much better how to solve this than I do.

**Comment by Tryneus** [comment]
Looking into this, thanks for debugging this, @danielmewes.

**Comment by Tryneus** [comment]
Fix is up in review 2837.

**Comment by Tryneus** [comment]
This has been merged to `next` in commit 3c04055.  Bumped the serialization version to 2.1 and added a separate function for deserializing old `wire_func_t`s.  Will be in release 2.1.
